circuit_knitting/cutting/cutting_experiments.py

```diff
@@ -409,7 +409,7 @@ def _remove_resets_in_zero_state(
         circuit = circuit.copy()
 
     # Keep up with which qubits have at least one non-reset instruction
-    active_qubits = set()
+    active_qubits: set[int] = set()
```

Comment:
Is this for clarity or was there some other reason?

Reply:
The linter complained once I made the early termination change until I added this annotation.  (I even reverted back to your original, just to double check the linter was fine with it on my machine, too.)